[v3] rewrite general

palmtree5 opened the pull request:
Rewrite of general.py for v3. Probably not completely done depending on what 26 wants to change in general (if anything) but it's a start

palmtree5 committed:
[v3] rewrite general

Rapptz commented on cogs/general/__init__.py:
```diff
@@ -0,0 +1,7 @@
+
+def setup(bot):
+    n = General(bot)
+    bot.add_listener(n.check_poll_votes, "on_message")
```
You can have `on_message` directly in the cog.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+
+        if outcome is True:
+            await ctx.send("{} You win {}!"
+                               "".format(red_choice.value, author.mention))
```
What is with the disgusting newline here?

palmtree5 committed:
[v3][general] rename message listener

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        if not user:
+            user = author
+
+        roles = [x.name for x in user.roles if x.name != "@everyone"]
```
`roles = [x.name for x in user.roles if not x.is_everyone()]`

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        roles = [x.name for x in user.roles if x.name != "@everyone"]
+
+        joined_at = self.fetch_joined_at(user, guild)
+        since_created = (ctx.message.timestamp - user.created_at).days
```
`Message.timestamp` no longer exists, use `Message.created_at`.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+    @commands.guild_only()
+    async def userinfo(self, ctx, *, user: discord.Member=None):
+        """Shows users's informations"""
+        author = ctx.message.author
```
There is `Context.author`.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+    async def userinfo(self, ctx, *, user: discord.Member=None):
+        """Shows users's informations"""
+        author = ctx.message.author
+        guild = ctx.message.guild
```
There is `Context.guild`

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        game = "Chilling in {} status".format(user.status)
+
+        if user.game is None:
+            pass
```
This statement is meaningless. Why are you even checking it?

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+
+        if user.game is None:
+            pass
+        elif user.game.url is None:
```
I think you meant `if user.game and user.game.name`

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+            game = "Streaming: [{}]({})".format(user.game, user.game.url)
+
+        if roles:
+            roles = sorted(roles, key=[x.name for x in guild.role_hierarchy
```
The first role in the hierarchy is the everyone role. You can remove it by doing something as basic as `guild.role_hierarchy[1:]`. I also don't know what this is supposed to be doing.

If you want to sort the user's roles by hierarchy the only thing you have to do is `sorted(member.roles)`.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        name = str(user)
+        name = " ~ ".join((name, user.nick)) if user.nick else name
+
+        if user.avatar_url:
```
`User.avatar_url` now returns the default avatar URL if one is not found. You might want to change this to `if user.avatar` instead.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+                      m.status == discord.Status.idle])
+        total_users = len(guild.members)
+        text_channels = len([x for x in guild.channels
+                             if x.type == discord.ChannelType.text])
```
`type` attribute no longer exists. You might have meant `isinstance`.

You can do e.g. `len(guild.text_channels)` to get the text channel count.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        total_users = len(guild.members)
+        text_channels = len([x for x in guild.channels
+                             if x.type == discord.ChannelType.text])
+        voice_channels = len(guild.channels) - text_channels
```
`len(guild.voice_channels)` or `len(guild.channels) - len(guild.text_channels)` from earlier.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        data.add_field(name="Voice Channels", value=voice_channels)
+        data.add_field(name="Roles", value=len(guild.roles))
+        data.add_field(name="Owner", value=str(guild.owner))
+        data.set_footer(text="Guild ID: " + guild.id)
```
Can't add `int` and `str`.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+                                passed))
+
+        colour = ''.join([choice('0123456789ABCDEF') for x in range(6)])
+        colour = int(colour, 16)
```
Just use `random.randint` here. e.g. `random.randint(0, 0xFFFFFF)`.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        except ValueError:
+            pos = 0
+
+        search_terms = "+".join([encode(s) for s in search_terms])
```
Honestly this entire function is fucked.

`aiohttp` supports `params` as a keyword argument and will automatically do this for you. I don't even know what's going on here.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        search_terms = "+".join([encode(s) for s in search_terms])
+        url = "http://api.urbandictionary.com/v0/define?term=" + search_terms
+        try:
+            async with aiohttp.request("GET", url) as r:
```
This was removed in aiohttp v2.0. Create a session and use it instead.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        try:
+            async with aiohttp.request("GET", url) as r:
+                result = await r.json()
+            if result["list"]:
```
Store this in a temporary variable instead of calling `__getitem__` repeatedly.

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+            if text[0].lower() == "stop":
+                await self.endpoll(message)
+                return
+        if not self.getPollByChannel(message):
```
Missing PEP-8.

palmtree5 replied:
Not completely clear on this one

EDIT: figured out you were referring to the function name being camel case as opposed to all lower, separated by underscores

palmtree5 committed:
[v3][general] escape_mass_mentions -> escape

Rapptz commented on cogs/general/general.py:
```diff
@@ -0,0 +1,433 @@
+        msg.remove(self.question)
+        self.answers = {}
+        i = 1
+        for answer in msg: # {id : {answer, votes}}
```
You probably meant `for i, answer in enumerate(msg, 1)`.

Rapptz reviewed (changes requested):
This is no good.

palmtree5 committed:
[v3][general] implement recommended changes

palmtree5 committed:
[v3][general] PEP-8 function naming conventions

palmtree5 commented:
Changes done

Rapptz commented on cogs/general/general.py:
```diff
@@ -201,16 +214,14 @@ def __init__(self, bot):
         if roles:
-            roles = sorted(roles, key=[x.name for x in guild.role_hierarchy
-                                       if x.name != "@everyone"].index)
+            roles = sorted(user.roles)
```
You probably want to just move this line to the top a bit.

First initialise:

`roles = sorted(user.roles)[1:]`

Then check if it exists and transform it:

```py
if roles:
  roles = ', '.join(x.name for x in roles)
```

palmtree5 committed:
[v3][general] change role handling in userinfo

Twentysix26 committed:
Let's knock off old habits

;)

Twentysix26 committed:
Removed polls

Twentysix26 committed:
Refactoring

Twentysix26 committed:
Renamed [p]guildinfo to [p]serverinfo

Guild is how servers are referenced in the API, the interface calls them servers.

Twentysix26 committed:
Guild -> Server

Twentysix26 committed:
Imports
